### PR TITLE
Hijack Kokkos c++11 flags in ArborX

### PR DESCRIPTION
**Pull request by aprokop**

### CODE REVIEW

File: cmake/FindKokkos.cmake [hunk 1]
```diff
@@ -99,6 +99,16 @@ if("${Kokkos_LIBRARY}" MATCHES "kokkoscore")
   list(INSERT PC_Kokkos_LDFLAGS ${index} "-lkokkosalgorithms;-lkokkoscontainers;-lkokkoscore")
 endif()
 
+if("${PC_Kokkos_CFLAGS_OTHER}" MATCHES "c\\+\\+11")
+  # Kokkos was configured in C++11 mode. As ArborX requires C++14, but picks up
+  # flags from Kokkos, hack that flag.
+  # NOTE: This is really only useful for for building ArborX against Trilinos
```
Comment (masterleinad) [inline cmake/FindKokkos.cmake]:
```suggestion
  # NOTE: This is really only useful for building ArborX against Trilinos
```

**Comment by masterleinad** [comment]
After the release, we don't export the version flag through `Cflags` anymore (at least for a regular `CMake` install). Is that a problem for you?

**Comment by aprokop** [comment]
> So far, I wasn't aware that we need to support linking against `Kokkos` through `Trilinos`. I am not sure what this means for the changes after the `Kokkos` release.

This is necessary for ExaWind, as it will have to build against both ArborX and Trilinos. So it ArborX is built against a separate version of Kokkos, multiple Kokkos headers are going to clash.

> After the release, we don't export the version flag through `Cflags` anymore (at least for a regular `CMake` install). Is that a problem for you?

I'm not sure. I guess I'll need to take a look at the new Kokkos installation to see what's up. Is current Kokkos master the right place to check, or is it in some unmerged branches?

**Comment by masterleinad** [comment]
No, it's the `develop` branch.

File: cmake/FindKokkos.cmake (hunk 1 above)
Comment (dalg24) [reply cmake/FindKokkos.cmake]: "a Trilinos"?

Comment (masterleinad) [reply cmake/FindKokkos.cmake]: "a Trilinos installation"

**Comment by aprokop** [comment]
I'm closing this PR as it does not work anyway with the Kokkos 3 update. We may still need to address this as part of #224.

**Pull request closed without merging**